### PR TITLE
BREAKING: `DENO_FUTURE=1` by default, or welcome to Deno 2.0

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit effectively turns Deno into Deno 2.0.

This is done by forcing `DENO_FUTURE=1` env var, that was available in
the past few months to try Deno 2 changes.

This commit contains several breaking changes scheduled for Deno 2:
- all deprecated JavaScript APIs are not available any more, mostly `Deno.*` APIs
- `window` global is removed
- FFI, WebGPU and FS APIs are now stable and don't require `--unstable-*` flags
- import assertions are no longer supported
- "bring your own node modules" is enabled by default

This is the first commit in a series that are scheduled before the Deno 2 release.

Follow up work is tracked in https://github.com/denoland/deno/issues/25241.

### CODE REVIEW

**Review by iuioiua — APPROVED** [review]
I wonder whether we really require the ability to have `DENO_FUTURE=0`. Either way, LGTM!

**Review by iuioiua — DISMISSED** [review]
Tests will need to be updated

**Comment by iuioiua** [comment]
We're running into this issue quite a bit:
```
error: Could not resolve "@denotest/esm-basic", but found it in a package.json. Deno expects the node_modules/ directory to be up to date. Did you forget to run `npm install`?
```
How should we approach folders that have a `package.json` but don't have the `node_modules` folder?

**Comment by iuioiua** [comment]
Should we just use `--node-modules-dir=false`?

**Comment by bartlomieju** [comment]
Down to one failing spec test: `specs::npm::exec_file_inside_npm_package`

**Comment by bartlomieju** [comment]
```
failures:
    check::package_json_basic
    check::package_json_fail_check
    check::package_json_with_deno_json
    compile::compile_npm_specifiers
    info::package_json_basic
    lsp::lsp_node_modules_dir
    lsp::lsp_npm_workspace
    node_unit_tests::_fs_appendFile_test
    node_unit_tests::_fs_close_test
    node_unit_tests::_fs_fdatasync_test
    node_unit_tests::_fs_fstat_test
    node_unit_tests::_fs_fsync_test
    node_unit_tests::_fs_ftruncate_test
    node_unit_tests::_fs_futimes_test
    node_unit_tests::_fs_writeFile_test
    node_unit_tests::_fs_write_test
    npm::child_process_fork_test
    npm::compare_globals
    npm::local_dir_resolves_symlinks
    npm::node_modules_import_check
    npm::node_modules_import_run
    npm::reload_info_not_found_cache_but_exists_remote
    repl::package_json_uncached_no_error
    run::_025_hrtime
    run::_025_reload_js_type_error
    run::_058_tasks_microtasks_close
    run::_078_unload_on_exit
    run::import_attributes_dynamic_import
    run::lock_deno_json_package_json_deps
    run::onload
    run::package_json_auto_discovered_for_npm_binary
    run::package_json_with_deno_json
    run::rejection_handled
    run::tls_connecttls
    run::tls_starttls
    run::webstorage_config_file
    run::webstorage_location_precedes_config
    run::webstorage_location_shares_origin
    run::webstorage_main_module
    run::webstorage_serialization
    task::task_both_package_json_selected
    task::task_package_json_npm_bin
    test::package_json_basic
    task::task_npx_on_own
```

**Comment by bartlomieju** [comment]
Remaining failures:
```
failures:
    lsp::lsp_node_modules_dir
    lsp::lsp_npm_workspace
    npm::local_dir_resolves_symlinks
    npm::node_modules_import_check
    npm::node_modules_import_run
    npm::reload_info_not_found_cache_but_exists_remote
    repl::package_json_uncached_no_error
    run::lock_deno_json_package_json_deps
    run::package_json_auto_discovered_for_npm_binary
    run::package_json_with_deno_json
    task::task_both_package_json_selected
    task::task_package_json_npm_bin
    test::package_json_basic

```

**Review by lucacasonato — COMMENTED** [review]
Can we mark these as `TODO(2.0)` or something so we can easially grep for them?

**Review by nathanwhit — APPROVED** [review]
LGTM, cleanup will be fun 😅

File: tests/integration/lsp_tests.rs [hunk 1]
```diff
@@ -14240,7 +14247,7 @@ fn lsp_deno_json_workspace_node_modules_dir() {
     "project1/deno.json",
     json!({
       "workspace": ["project2"],
-      "nodeModulesDir": true,
+      "nodeModules": "local-auto",
```
Comment (dsherret) [inline tests/integration/lsp_tests.rs:14250]: I think we should change this to `"nodeModules": "auto"`. This is too much to type.

**Review by dsherret — APPROVED** [review]
LGTM!

**Pull request merged**